cirq-superstaq/cirq_superstaq/job.py

```diff
@@ -129,7 +131,7 @@ def num_qubits(self) -> int:
             SuperstaqUnsuccessfulJobException: If the job failed or has been canceled or deleted.
             SuperstaqException: If unable to get the status of the job from the API.
         """
-        self._check_if_unsuccessful()
+        self._refresh_job()
```

Comment:
Since `_refresh_job()` calls the client each time, is it worth storing fixed data (like `num_qubits`) in an attribute like `job_info` at the start?

Reply:
great idea. will update to refresh iff the requested field isn't already in `self._job`